src/main/java/duke/logic/InputValidator.java

```diff
@@ -0,0 +1,199 @@
+        }
+        return true;
+    }
+
+    /**
+     * Validates that the instructions have correct Date and Time formats.
+     * Date and Time exists as <code>Strings</code> in the array.
+     * They are extracted and split into their constituent <code>Integer</code> values and parsed.
+     * If the parse fails, an error is raised.
+     * Else, they are put through validation to ensure it is proper Date and Time values.
+     *
+     * @param instructionArray <code>String</code> containing length of user instruction.
+     * @param index            <code>Integer</code> containing location of separator.
+     * @return boolean denoting the validation results
+     * @throws InvalidInstructionFormatException If validation fails
+     */
+    public static boolean validateDateAndTime(String[] instructionArray, int index)
+            throws InvalidInstructionFormatException {
+
+        String date = instructionArray[index + 1];
+        String time = instructionArray[index + 2];
+
+        // INPUT DATE FORMAT: DD/MM/YYYY
+        // INPUT TIME FORMAT: hh/mm/ss
+        int year;
+        int month;
+        int day;
+        int hour;
+        int minute;
+        int second;
+
+        String[] dateArray = date.split("/");
+        String[] timeArray = time.split("/");
+
+        if (dateArray.length != 3 || timeArray.length != 3) {
+            throw new InvalidInstructionFormatException();
+        }
+
+        try {
+            day = Integer.parseInt(dateArray[0]);
+            month = Integer.parseInt(dateArray[1]);
+            year = Integer.parseInt(dateArray[2]);
+
+            hour = Integer.parseInt(timeArray[0]);
+            minute = Integer.parseInt(timeArray[1]);
+            second = Integer.parseInt(timeArray[2]);
+        } catch (NumberFormatException nfe) {
+            throw new InvalidInstructionFormatException();
+        }
+
+        // input validation for time
+        if (hour < 0 || hour > 23 || minute < 0 || minute > 59 || second < 0 || second > 59) {
+            throw new InvalidInstructionFormatException();
+        }
+
+        // input validation for date
+        if (year <= 0 || month <= 0 || month > 12 || day <= 0) {
+            throw new InvalidInstructionFormatException();
+        }
+
+        switch (month) {
+        case 1:
+        case 3:
+        case 5:
+        case 7:
+        case 8:
+        case 10:
+        case 12:
+            if (day > 31) {
+                throw new InvalidInstructionFormatException();
+            }
+            break;
+        case 4:
+        case 6:
+        case 9:
+        case 11:
+            if (day > 30) {
+                throw new InvalidInstructionFormatException();
+            }
+            break;
+        default:
+            if ((CommonMethod.isLeapYear(year) && day > 29) || (!CommonMethod.isLeapYear(year) && day > 28)) {
+                throw new InvalidInstructionFormatException();
+            }
+        }
```

Comment:
Consider using in built Date and Time objects that are provided by the Java JDK! They provide extensive functionality including parsing and Formatting abilities!